public/javascript/pump.js

```diff
@@ -669,22 +658,53 @@ if (!window.Pump) {
     };
 
     Pump.ajaxError = function(jqXHR, textStatus, errorThrown) {
-        Pump.error(Pump.jqxhrError(jqXHR));
+        Pump.jqxhrError(jqXHR);
     };
 
     Pump.jqxhrError = function(jqxhr) {
         var type = jqxhr.getResponseHeader("Content-Type"),
             response;
-        if (type && type.indexOf("application/json") !== -1) {
+
+        if (_.isObject(jqxhr.responseJSON) &&
+            _.isString(jqxhr.responseJSON.error)) {
+            Pump.error(new Error(jqxhr.responseJSON.error));
+        } else if (type && type.indexOf("application/json") !== -1) {
             try {
                 response = JSON.parse(jqxhr.responseText);
                 Pump.error(new Error(response.error));
             } catch (err) {
                 Pump.error(new Error(jqxhr.status + ": " + jqxhr.statusText));
             }
-        } else {
+        } else if (jqxhr.status && jqxhr.statusText) {
             Pump.error(new Error(jqxhr.status + ": " + jqxhr.statusText));
+        } else {
+            Pump.error(jqxhr);
+        }
+    };
+
+    Pump.getError = function(err) {
+
+        if (_.isString(err)) {
+            return err;
+        } else if (_.isObject(err)) {
+
+            if (err.stack) {
+                console.log(err.stack);
```

Comment:
This should probably be `return err.stack;`, no?

Reply:
No, in this case, should return the error message to the user, the stack is more descriptive (example: internal server error in node_modules/... etc) and should be a string to display in the alert banner